pkg/postage/postagecontract/contract.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package postagecontract
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/postage/listener"
+	"github.com/ethersphere/bee/pkg/settlement/swap/transaction"
+	"github.com/ethersphere/sw3-bindings/v2/simpleswapfactory"
+)
+
+var (
+	postageStampABI   = parseABI(listener.PostageStampABIJSON)
+	erc20ABI          = parseABI(simpleswapfactory.ERC20ABI)
+	batchCreatedTopic = postageStampABI.Events["BatchCreated"].ID
+
+	ErrBatchCreate = errors.New("batch creation failed")
+)
+
+type Interface interface {
+	CreateBatch(ctx context.Context, initialBalance *big.Int, depth uint8) ([]byte, error)
+}
+
+type postageContract struct {
+	owner               common.Address
+	postageStampAddress common.Address
+	bzzTokenAddress     common.Address
+	transactionService  transaction.Service
+	postageService      postage.Service
+}
+
+func New(
+	owner common.Address,
+	postageStampAddress common.Address,
+	bzzTokenAddress common.Address,
+	transactionService transaction.Service,
+	postageService postage.Service,
+) Interface {
+	return &postageContract{
+		owner:               owner,
+		postageStampAddress: postageStampAddress,
```

Comment:
this is the contract address no? please do not call it stamp address

Reply:
see https://github.com/ethersphere/bee/pull/1142/files#r564473904